spec/rester/client_spec.rb

```diff
@@ -34,28 +33,15 @@ def json_h(params)
     # Response Hash
     let(:res_hash) { req_hash.map{|k,v| [k, v.nil? ? nil : v.to_s]}.to_h }
 
-    describe '#connect', :connect do
-      subject { client.connect(url) }
-
-      context 'with valid url' do
-        let(:url) { test_url }
-
-        it 'should return nil' do
-          expect(subject).to be nil
-        end
-      end # valid url
-    end # #connect
-
     describe '#connected?', :connected? do
       subject { client.connected? }
+      it { is_expected.to be true }
 
-      context 'before connecting' do
-        it { is_expected.to be false }
-      end
-
-      context 'after connecting' do
-        before { client.connect(test_url) }
-        it { is_expected.to be true }
+      context 'with connection error' do
+        let(:adapter) { Client::Adapters::HttpAdapter.new }
+        it {
+          expect { subject }.to raise_error Errors::ConnectionError
+        }
```

Comment:
Also, how about instead:

```ruby
context 'with adapter not connected' do
  let(:adapter) { double('adapter', connected?: false) }
  it { expect { subject }.to raise_error Errors::ConnectionError }
end
```

That way we're testing independently of how `HttpAdapter` behaves.